vcr/vcr.go

```diff
@@ -224,8 +227,62 @@ func (c *vcr) Resolve(ID string) (did.VerifiableCredential, error) {
 	panic("implement me")
 }
 
-func (c *vcr) Verify(vc did.VerifiableCredential, credentialSubject interface{}, at time.Time) (bool, error) {
-	panic("implement me")
+func (c *vcr) Verify(vc did.VerifiableCredential, at time.Time) error {
+	// it must have valid content
+	validator, _ := credential.FindValidatorAndBuilder(vc)
+	if validator == nil {
+		return errors.New("unknown credential type")
+	}
+
+	if err := validator.Validate(vc); err != nil {
+		return err
+	}
+
+	// create correct challenge for verification
+	payload, err := generateCredentialChallenge(vc)
+	if err != nil {
+		return fmt.Errorf("cannot generate challenge: %w", err)
+	}
+
+	// extract proof, can't fail already done in generateCredentialChallenge
+	var proofs = make([]JSONWebSignature2020Proof, 0)
+	_ = vc.UnmarshalProofValue(&proofs)
+	proof := proofs[0]
+	splittedJws := strings.Split(proof.Jws, "..")
+	if len(splittedJws) != 2 {
+		return errors.New("invalid 'jws' value in proof")
+	}
+	sig, err := base64.RawURLEncoding.DecodeString(splittedJws[1])
+	if err != nil {
+		return err
+	}
+
+	// find key
+	pk, err := c.docResolver.ResolveSigningKey(proof.VerificationMethod.String(), &at)
+	if err != nil {
+		return err
+	}
+
+	// the proof must be correct
+	verifier, _ := jws.NewVerifier(jwa.ES256)
```

Comment:
theoretically we support other keys in our DIDs than just RSA so should be check that?

Reply:
added a `SigningAlgorithm` func to crypto/jwx to determine signing algo based on key.